### PR TITLE
MINOR: code cleanup for Kafka Streams task classes

### PR DESCRIPTION
**Pull request by mjsax**
Not functional change. Pure code cleanup

Call for review @vvcephei

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java [hunk 1]
```diff
@@ -16,82 +16,85 @@
...
     private Task.State state = CREATED;
-    protected Set<TopicPartition> inputPartitions;
-    protected ProcessorTopology topology;
 
     protected final TaskId id;
+    protected final ProcessorTopology topology;
     protected final StateDirectory stateDirectory;
     protected final ProcessorStateManager stateMgr;
 
+    protected Set<TopicPartition> inputPartitions;
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java:40]: Align members to constructor parameter order, and group `final` / mutable.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java [hunk 2]
```diff
@@ -16,82 +16,85 @@
...
                  final StateDirectory stateDirectory,
                  final ProcessorStateManager stateMgr,
                  final Set<TopicPartition> inputPartitions) {
         this.id = id;
-        this.stateMgr = stateMgr;
         this.topology = topology;
-        this.inputPartitions = inputPartitions;
+        this.stateMgr = stateMgr;
         this.stateDirectory = stateDirectory;
+        this.inputPartitions = inputPartitions;
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java:51]: Align assignment to parameter order.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java [hunk 3]
```diff
@@ -97,13 +97,9 @@ public boolean isValidTransition(final State newState) {
         }
     }
 
-    TaskId id();
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java:100]: Group and order methods (compare in-line comments). -- Sub-classed inherit this ordering.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java [hunk 4]
```diff
@@ -97,13 +97,9 @@ public boolean isValidTransition(final State newState) {
...
 
-    State state();
 
-    boolean isActive();
-
-    boolean isClosed();
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java:106]: This method is unused and removed. That is the only actual change.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java [hunk 5]
```diff
@@ -116,89 +112,100 @@ public boolean isValidTransition(final State newState) {
...
+    TaskId id();
+
+    State state();
+
+    boolean isActive();
+
+    /**
+     * Updates input partitions after a rebalance
+     */
+    void updateInputPartitions(final Set<TopicPartition> topicPartitions, final Map<String, List<String>> nodeToSourceTopics);
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java:191]: Renamed from `update` to `updateInputPartitions`

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java [hunk 6]
```diff
@@ -959,28 +992,23 @@ private void maybeRecordE2ELatency(final long recordTimestamp, final long now, f
...
-     * Whether or not a request has been made to commit the current state
-     */
-    @Override
-    public boolean commitRequested() {
-        return commitRequested;
+    public boolean hasRecordsQueued() {
+        return numBuffered() > 0;
     }
 
+    // visible for testing
```
Comment (vvcephei) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java:1003]: Can we avoid these comments? I've come across too many cases where it had become untrue recently, to the point where it just seems pointless to have them.

I know you're preserving the comment that was there before while moving the method, but I think we should just clean up the comments as well.

**Comment by mjsax** [comment]
Closing in favor of #8852

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java [hunk 7]
```diff
@@ -116,89 +112,100 @@ public boolean isValidTransition(final State newState) {
...
+
+    Set<TopicPartition> inputPartitions();
+
+    /**
+     * @return any changelog partitions associated with this task
+     */
+    Collection<TopicPartition> changelogPartitions();
+
+
+    // IQ related methods
```
Comment (vvcephei) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java:201]: I appreciate that you've taken the time to organize these methods, and that these comments are an attempt to make sure they stay organized, but I'm afraid that they'll just become misleading over time, the way that the "for testing" comments do.

**Review by vvcephei — COMMENTED** [review]
Ah. You closed it right before I submitted my review ;)

**Pull request closed without merging**